src/main/java/com/icthh/xm/ms/entity/service/privileges/custom/EntityCustomPrivilegeService.java

```diff
@@ -0,0 +1,198 @@
+            if (customPrivileges.containsKey(privilegesExtractor.getSectionName())) {
+                customPrivileges.get(privilegesExtractor.getSectionName()).addAll(value);
+            } else {
+                customPrivileges.put(privilegesExtractor.getSectionName(), value);
+            }
+        }
+        privileges.putAll(customPrivileges);
+    }
+
+    private Map<String, List<Map<String, Object>>> getPrivilegesConfig(
+        Configuration customPrivileges) throws java.io.IOException {
+        Map<String, List<Map<String, Object>>> privileges = new HashMap<>();
+        if (isConfigExists(customPrivileges)) {
+            privileges = mapper.readValue(customPrivileges.getContent(), new TypeReference<Map<String, List<Object>>>() {
+            });
+        }
+        return privileges;
+    }
+
+    private String sha1Hex(Configuration configuration) {
+        return ofNullable(configuration).map(Configuration::getContent).map(DigestUtils::sha1Hex).orElse(null);
+    }
+
+    private boolean isConfigExists(Configuration configuration) {
+        return ofNullable(configuration)
+            .map(Configuration::getContent)
+            .map(StringUtils::isNotBlank)
+            .orElse(false);
+    }
+
+    @SneakyThrows
+    private void setNewPermissionsDefaultValue(Map<String, TypeSpec> specs,
+                                               String tenantKey,
+                                               String permissionsSpecPath,
+                                               Configuration permissionsSpec) {
+
+        val permissions = updatePermissions(permissionsSpec, specs, tenantKey);
+        String permissionsYml = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(permissions);
+        Configuration configuration = new Configuration(permissionsSpecPath, permissionsYml);
+        if (DigestUtils.sha1Hex(permissionsYml).equals(sha1Hex(permissionsSpec))) {
+            log.info("Permissions configuration not changed");
+            return;
+        }
+        commonConfigRepository.updateConfigFullPath(configuration, sha1Hex(permissionsSpec));
+    }
+
+    private Map<String, ?> updatePermissions(Configuration permissionConfiguration,
+                                             Map<String, TypeSpec> spec,
+                                             String tenantKey) {
+        val permissions = getPermissionsConfig(permissionConfiguration);
+
+        for(CustomPrivilegesExtractor privilegesExtractor: privilegesExtractors) {
+            DefaultPrivilegesValue defaultValue = privilegesExtractor.getDefaultValue();
+            if (defaultValue == NONE) {
+                continue;
+            }
+
+            String sectionName = privilegesExtractor.getSectionName();
+            permissions.computeIfAbsent(sectionName, key -> new TreeMap<>());
+            val permission = permissions.getOrDefault(sectionName, new TreeMap<>());
```

Comment:
don't need to do `get` operation. The right value is returned by `computeIfAbsent`